### ISSUE TITLE
No magnetometer/compass readings

### ISSUE THREAD

**Issue by Blackoverflow** [HelloVolla/ubuntu-touch-beta-tests#75]
**Details about installed OS version**
Ubuntu Touch OTA-14

**Describe the bug**
The app "SensorStatus" doesn't report readings for the menu items "Magnetometer" and "Compass".
The app "Xompass" doesn't work.
The app "stellarium" isn't able to orient itself based on compass readings.

**Expected behavior**
The Volla Phone is equipped with a geomagnetic sensor.
I wish to be able to use its readings.

**Comment by JamiKettunen** [comment]
No app indeed seems to be able to report readings from this sensor, however it is listed under `test_sensors` so perhaps it can be enabled for this port in the future:
```
=== Sensor 1 ==
Name: MAGNETOMETER
Vendor: MTK
Version: 0x1
Handle: 0x2
Type: 2
maxRange: 4912.0
resolution: 0.150000
power: 0.0 mA
minDelay: 20000
```

**Comment by Flohack74** [comment]
Its an "upstream" problem: The code for additional sensors is simply not on. Moreover, Compass is a derived sensor, so it needs to be calculated. Please see my work here: https://github.com/ubports/platform-api/pull/8 and https://github.com/ubports/qtubuntu-sensors/pull/4

**Comment by Flohack74** [comment]
I propose to merge this in a next OTA, however my compass calculation is still broken ;) but some sensors will come to life with that.

**Comment by gbdomubpkm** [comment]
@Flohack74 Should Xompass work in RC 43 on the Volla ? Because it seems not to me.

**Comment by Flohack74** [comment]
@gbdomubpkm I do not own a Volla, but I got some readings on other devices. We need to double-check with someone else maybe.

**Comment by gbdomubpkm** [comment]
> some readings on other device

Does it mean it works on RC 43 on other Halium 9 devices ?

**Comment by Flohack74** [comment]
@gbdomubpkm it works on devel for my Oneplus 5T and Pixel 2XL I think, yes

**Comment by gbdomubpkm** [comment]
Ok. I'm not fully sure if compass works in this RC 43 for halium 9 devices : If compass works in this RC, it is the .click Xompass that doesn't work on Volla. Or maybe compass hasn't been pushed in RC 43, only yet in devel,  so wait next RC. Thanks.

**Comment by Flohack74** [comment]
Xompass is broken AFAIK, did you try with sensorsstatus or Messwerk?

**Comment by gbdomubpkm** [comment]
Yes Xompass probably broken. Compass in SensorsStatus probably works indeed but in Messwerk, i don't see the name 'compass'.
But it is only in Xompass that is really speaking with the needle that should move.

**Comment by Flohack74** [comment]
Xompass only works on some legacy devices of BQ I think, since it uses the hardware driver directly. I think thats the issue, but it should be easy to make a small QML app with a rotating needle that attaches to the QML sensor API.

**Comment by gbdomubpkm** [comment]
From the user's point of view, the only thing that matters is to see a needle that indicates North so as not to lose it 😉

**Comment by Flohack74** [comment]
@gbdomubpkm well then I need to ask to remove Xompass since it does not fit for 90% of our devices. See also the note:
![grafik](https://user-images.githubusercontent.com/16186470/147546761-30c6af9a-dfd5-44fb-b692-70bb702d170a.png)

**Comment by Flohack74** [comment]
You see it goes directly into a system node, instead of using the proper sensor API: _Unrestricted read access to: /sys/bus/platform/drivers/msensor/sensordata_ - well thats the wrong way to do it. It needs complete rework to use the Qt sensor API now.

**Comment by gbdomubpkm** [comment]
@Flohack74 christmas gift ☺ https://open-store.io/app/xompass.emanuelesorce

**Comment by JamiKettunen** [comment]
I suppose this is solved now (minus a specific app)?

**Comment by Flohack74** [comment]
Phew @JamiKettunen I do not believe this is really solved, I got never a proof that modifications to sensorfw were really successful in giving us a more or less good reading (not talking about precise ofc)

**Comment by Flohack74** [comment]
I just checked on the SHIFt 6mq and the Fairphone 4 and on both devices the values are jumping, not consistent and very unbplausible.
That would need deeper inspection why sensorfw compass filter/sensor does not work on UT
